### PR TITLE
Add eslint settings to vscode config

### PR DESCRIPTION
**Pull request by moxian**
## What Does This PR Do
Adds eslint extension settings to the vscode project config.

## Why It's Good For The Game
I've spent way too much time using my default js formatter extension and getting increasingly annoyed by the fact that its output is different from the one required by eslint, and that i need to remember to `yarn run lint --fix` manually before PR submission.

So i delved into this.
Turns out we have eslint extension in the recommended ones. And i have it installed. But it just doesn't format shit. It's not even an option in the formatter selection. And if you force it to be used anyways, then it fails due to being unable to find `react` module.
Well, that's all because it treats the Paradise/ root as a root of the js project and expects to find `node_modules/` right below it, as any sane JS project would do. But we are not a js project, and all our js stuff lives under `tgui/`. So eslint needs to be told that:
```js
	"eslint.workingDirectories": [
		"tgui/"
	]
```

Well, that worked! But that also meant that any styling issues like a wrong amount of whitespace immediately gave me red squiggles, which is also annoying since that's not a hard error. Except we *want* that to be a hard error for CI purposes... So we have two options basically: either downgrade the severity in vscode, OR downgrade the severity of everything in `.eslintrc` + upgrade it back in the CI via `--max-warnings=0`. I opted for the former as that's less work in the short term.
```js
	"eslint.rules.customizations": [
		{ "rule": "*", "severity": "warn" }
	]
```

So, yay, we have the yellow squigglies. And with one more change, we also have the vscode formatting support:
```js
    "eslint.format.enable": true,
    "[javascript]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
```

## Images of changes
20220427-155926-Code

### CODE REVIEW

**Comment by moxian** [comment]
CI dislikes my comments in `.vscode\settings.json` (which is actually [json5](https://json5.org/), and not a real json). I'm really tempted to just exclude it from the CI check, but if that is not in accord with Maint's will, i'll just  remove the comments.

Please advise.

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,5 +6,20 @@
 	"editor.guides.bracketPairs": "active",
 	"gitlens.advanced.blame.customArguments": [
 		"--ignore-revs-file", "${workspaceRoot}/.git-blame-ignore-revs"
-	]
+	],
+	// ESLint settings:
```
Comment (AffectedArc07) [inline .vscode/settings.json:10]:
```suggestion
	"paradisegit.comment1": "ESLint settings",
```

File: .vscode/settings.json [hunk 2]
```diff
@@ -6,5 +6,20 @@
...
+	"eslint.workingDirectories": [
+		"tgui/"
+	],
+	"eslint.rules.customizations": [
+		// We really want to fail the CI builds on styling errors,
+		// but it's better to show them as yellow squigglies in IDE
+		// and thus differentiate from the red typescript ones which
+		// are actually hard errors.
```
Comment (AffectedArc07) [inline .vscode/settings.json:18]:
```suggestion
		"paradisegit.comment2": [
			"We really want to fail the CI builds on styling errors,",
			"but it's better to show them as yellow squigglies in IDE",
			"and thus differentiate from the red typescript ones which",
			"are actually hard errors.",
		],
```

**Review by AffectedArc07 — COMMENTED** [review]
Its hacky but it will pass CI

File: .vscode/settings.json (hunk 2 above)
Comment (moxian) [reply .vscode/settings.json:18]: That is indeed another way to address it.
Do you prefer that to adding `-not -path ".vscode/*"` into `ci.yml`?

Comment (AffectedArc07) [reply .vscode/settings.json:18]: You could also do that, but I feel like it should still be validated

**Comment by moxian** [comment]
Ok, took the fourth option, as you suggested on discord, and validated the comment-full json as such. It works, and is non-ugly AFAICT.

The dumpster fire of force pushes above is due to me seeing that the lint CI now runs 10 seconds slower, panicking ("i bet the tool i used is just garbage then!!!"), and trying to do *something* to fix it back. But turns out that's just github being a github, as our jsons are so tiny, that the entire validation takes well under a second.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**